OpenRA.Mods.Common/Warheads/CreateResourceWarhead.cs

```diff
@@ -46,13 +49,17 @@ public override void DoImpact(Target target, WarheadArgs args)
 				Log.Write("debug", "Warhead defines an invalid resource type '{0}'".F(AddsResourceType));
 			else
 			{
-				foreach (var cell in allCells)
+				var resLayers = world.WorldActor.TraitsImplementing<ResourceLayer>();
+				foreach (var rl in resLayers)
 				{
-					if (!resLayer.CanSpawnResourceAt(resourceType, cell))
-						continue;
+					foreach (var cell in allCells)
+					{
+						if (!rl.CanSpawnResourceAt(resourceType, cell))
+							continue;
 
-					var splash = world.SharedRandom.Next(1, resourceType.Info.MaxDensity - resLayer.GetResourceDensity(cell));
-					resLayer.AddResource(resourceType, cell, splash);
+						var splash = world.SharedRandom.Next(1, resourceType.Info.MaxDensity - rl.GetResourceDensity(cell));
+						rl.AddResource(resourceType, cell, splash);
```

Comment:
We should have a `break` after this to avoid spawning resources in multiple layers.

Reply:
Hm, after trying to implement that, I'm not so sure; we can't break from looping over the cells (otherwise the other cells are ignored), but if we break from the resource layer loop once we added some resource to this layer, we ignore other layers entirely, even if some of those cells belong to other layers.

Caching each cell that received resources to exclude them when looping over the next layer sounds a bit overkill, too (and a potential performance hit), so shouldn't we leave it to `ResourceLayer.CanSpawnResourceAt` to check if the cell belongs to a different layer already?

Reply:
On second thought, multiple res layers shouldn't share the same ResourceTypes though, so I guess breaking from the res layer loop is fine.